api_tests/tokens/test_views.py

```diff
@@ -0,0 +1,277 @@
+import copy
+import mock
+
+from nose.tools import *  # flake8: noqa
+
+from website.models import User, ApiOAuth2PersonalToken
+from website.util import api_v2_url
+from website.util import sanitize
+
+from tests.base import ApiTestCase
+from tests.factories import ApiOAuth2PersonalTokenFactory, AuthUserFactory
+
+
+def _get_token_detail_route(token):
+    path = "tokens/{}/".format(token._id)
+    return api_v2_url(path, base_route='/')
+
+
+def _get_token_list_url():
+    path = "tokens/"
+    return api_v2_url(path, base_route='/')
+
+
+class TestTokenList(ApiTestCase):
+    def setUp(self):
+        super(TestTokenList, self).setUp()
+
+        self.user1 = AuthUserFactory()
+        self.user2 = AuthUserFactory()
+
+        self.user1_tokens = [ApiOAuth2PersonalTokenFactory(owner=self.user1, user_id=self.user1._id) for i in xrange(3)]
+        self.user2_tokens = [ApiOAuth2PersonalTokenFactory(owner=self.user2, user_id=self.user2._id) for i in xrange(2)]
+
+        self.user1_list_url = _get_token_list_url()
+        self.user2_list_url = _get_token_list_url()
+
+        self.sample_data = {
+            'data': {
+                'type': 'tokens',
+                'attributes': {
+                    'name': 'A shiny new token',
+                    'scopes': 'osf.full_write',
+                    'owner': 'Value discarded',
+                    'token_id': 'Value discarded',
+                }
+            }
+        }
+
+    def test_user1_should_see_only_their_tokens(self):
+        res = self.app.get(self.user1_list_url, auth=self.user1.auth)
+        assert_equal(len(res.json['data']),
```

Comment:
In general, we're moving towards encouraging all api tests to check the return code before checking anything else. (it makes it slightly easier to tell why a test is failing)